server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -375,6 +375,34 @@ protected int checkHasMutuallyExclusiveOptions(String options, List<TerminalNode
         }
         return nodes.size();
     }
+    /**
+     * Flags errors for rule lists passed as parameters if there are multiple instances of mutually
+     * exclusive options.
+     *
+     * @param rules   Lists of TerminalNode to iterate through
+     * @return Number of TerminalNode instances found
+     */
+    protected int checkHasMutuallyExclusiveOptions(List<TerminalNode>... rules) {
+        List<TerminalNode> nodes =
+                Stream.of(rules)
+                        .filter(rule -> !rule.isEmpty())
+                        .flatMap(Collection::stream)
+                        .collect(Collectors.toList());
+        nodes.removeIf(Objects::isNull);
+
+        if (!nodes.stream()
+                .allMatch(e -> e.getSymbol().getType() == nodes.get(0).getSymbol().getType())) {
+            nodes.forEach(
+                    node -> {
+                        throwException(
+                                ErrorSeverity.ERROR,
+                                getLocality(node),
+                                "Exactly one option required, options are mutually exclusive: ",
+                               nodes.toString().replace(",", " or").replace("[", "").replace("]", ""));
```

Comment:
Alright, but if I specify `TOKEN1 TOKEN1 TOKEN2 TOKEN2` the message will contain the duplicates. So only distinct tokens should be reported.

Reply:
Added at : [1e31316](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2666/commits/1e31316da2e63a9e38d39371e54e7f7258f2ed93)

Reply:
does it work with mixed case input?

Reply:
could you elaborate the mixed case input ?